### PR TITLE
feat(sweeps): Adds Sweep.expected_run_count to public Api

### PR DESCRIPTION
**Pull request by gtarpenning**
https://wandb.atlassian.net/browse/WB-11414

Description
-----------
Adds expected run count -- now with a legacy fallback query 

Testing
-------
added testing in `test_public_api.py`

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/apis/public.py [hunk 1]
```diff
@@ -2605,8 +2628,17 @@ def get(
...
+        response = None
+        try:
+            response = client.execute(query, variable_values=variables)
+        except Exception:
+            # Don't handle exception, rely on legacy query
+            query = cls.LEGACY_QUERY
+            response = client.execute(query, variable_values=variables)
+
+        if response is None:
+            return None
```
Comment (KyleGoyette) [inline wandb/apis/public.py:2640]: Im surprised we need this check now, do you know why this is required?

File: wandb/apis/public.py [hunk 2]
```diff
@@ -2605,8 +2628,17 @@ def get(
         }
         variables.update(kwargs)
 
-        response = client.execute(query, variable_values=variables)
-        if response.get("project") is None:
+        response = None
+        try:
+            response = client.execute(query, variable_values=variables)
+        except Exception:
```
Comment (KyleGoyette) [inline wandb/apis/public.py:2634]: it feels not great to fall back on any exception, but I think this makes sense. We just end up falling back to the original query and if the exception was unrelated to the original query structure, the next query would just raise it.

But we should maybe have a TODO to find a better way to do this in the future

File: wandb/apis/public.py (hunk 1 above)
Comment (gtarpenning) [reply wandb/apis/public.py:2640]: `None.get()` excepts right?

Comment (gtarpenning) [reply wandb/apis/public.py:2640]:
```
>>> response = None
>>> response.get("project")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```

**Review by KyleGoyette — APPROVED** [review]

**Pull request merged**